text/0000-support-elastic-common-schema-in-opentelemetry.md

```diff
@@ -0,0 +1,259 @@
+
+Adding the coverage of ECS to OTEL would provide guidance to authors of OpenTelemetry Collector Logs Receivers and help establish the OTEL Collector as a de facto standard log collector with a well-defined schema to allow for richer data definition.
+
+In addition to the use case of structured logs, the maturity of ECS for SIEM (Security Information and Event Management) is a great opportunity for OpenTelemetry to expand its scope to the security use cases.
+
+Another significant use case is providing first-class support for Kubernetes application logs, system logs as well as application introspection events. We would also like to see support for structured events (e.g. [k8seventsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8seventsreceiver)) and using ‘content-type' to identify event types.
+
+We'd like to see different categories of structured logs being well-supported in the [OTEL Log Data Model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md), presumably through [semantic conventions for log attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-attributes). For example, NGINX access logs and Apache access logs should be processed the same way as structured logs. This would help in trace and metric correlation with such log data as well as it would help grow the ecosystem of curated UIs provided by observability backends and monitoring dashboards (e.g. one single HTTP Access log dashboard benefiting Apache HTTPD, Nginx, and HAProxy).
+
+
+## Customer Motivation
+
+Adoption of OTEL logs will accelerate greatly if ECS is leveraged as the common standard, using this basis for normalization. OTEL Logs adoption will be accelerated by this support. For example, ECS can provide the unified structured format for handling vendor generated along with open source logs.
+
+Customers will benefit from turnkey logs integrations that will be fully recognized by OTEL compatible observability products and services.
+
+OpenTelemetry logging is today mostly structured when instrumentation libraries are used. However, most of the logs which exist today are generated by software, hardware, and cloud services which the user cannot control. OpenTelemetry provides a limited set of "reference integrations" to  structure logs: primarily the [OpenTelemetry Collector Kubernetes Events Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8seventsreceiver) and an example of a regexp based parsing of Tomcat access logs with OpenTelemetry Collector File Receiver ([here](https://github.com/open-telemetry/opentelemetry-log-collection/blob/30807b96b2f0771e7d11452ebf98fe5e211ed6d7/examples/tomcat/config.yaml#L20)).
```

Comment:
It is not clear to me how this proposal to add ECS fields in the OTel semconv would address the challenge described here. Can you expand on this?

Reply:
The purpose of ECS is to provide correlation between different log formats. For example, if you are looking for a log about a dropped firewall event across multiple firewalls. The log message is represented in a similar form in ECS versus the way that Linux, Cisco, and Palo Alto might format the firewall log. This allows us to correlate data with a single query versus trying to build complex queries (and parsing rules) on the backend to correlate. This purpose of this is not for otel instrumented code, but for non-instrumented code.

Reply:
Thank you for the explanation. The expectation is not that such vendors would pick up ECS or OTel semconv as their logging format in any foreseeable future yet (and certainly not for already deployed appliances) but that we would be able to define transformation/mapping rules that are then implemented in the respective OTel collector receivers or processors, right?
Do such mapping rules for common (log) sources/formats already exist today in ECS so they can then be used in OTel or are they yet to be defined?

Reply:
@jkowall @cyrille-leclerc Can you please address the questions above?

Reply:
It makes it much easier to have a schema to normalize the fields and meanings together for correlation. Ultimately, we would want all vendors to log in Otel natively, but that will take a long time. You can do the correlation and parsing in various stages of the pipeline. It might be done in the collector, or as part of an ingestion pipeline to a telemetry service. We see both methods at my organization since we have a sophisticated parsing pipeline. 

There are mapping rules for ECS inside some solutions like Logstash, @cyrille-leclerc maybe you can suggest other implementations of ECS and parsing rules?

Reply:
Resolving this conversation as @jkowall answered